pkg/apis/authorization/types.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authorization
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
+// spec.namespace means "in all namespaces".
+type SubjectAccessReview struct {
+	unversioned.TypeMeta
+
+	// Spec holds information about the request being evaluated
+	Spec SubjectAccessReviewSpec
+
+	// Status is filled in by the server and indicates whether the request is allowed or not
+	Status SubjectAccessReviewStatus
+}
+
+// SelfSubjectAccessReview checks whether or the current user can perform an action.  Not filling in a
+// spec.namespace means "in all namespaces".  Self is a special case, because users should always be able
+// to check whether they can perform an action
+type SelfSubjectAccessReview struct {
+	unversioned.TypeMeta
+
+	// Spec holds information about the request being evaluated.
+	Spec SelfSubjectAccessReviewSpec
+
+	// Status is filled in by the server and indicates whether the request is allowed or not
+	Status SubjectAccessReviewStatus
+}
+
+// LocalSubjectAccessReview checks whether or not a user or group can perform an action in a given namespace.
+// Having a namespace scoped resource makes it much easier to grant namespace scoped policy that includes permissions
```

Comment:
I am confused about why the non-local version allows you to specify a namespace, if this one exists?